### PR TITLE
Update for German translation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
QT linguist changed ts version. Should I change it back?

**Comment by RaspberryPiFan** [comment]
I'm going to add more translation tomorow.

**Comment by ghost** [comment]
Should I translate PINN options?

**Comment by ghost** [comment]
image

Some text doesn't fit into his position
See issue https://github.com/procount/pinn/issues/303

**Comment by ghost** [comment]
Here's my fix for this:

**Comment by ghost** [comment]
image
It looks like that.

**Comment by ghost** [comment]
I can't translate this:
` 
 <message>
        <location filename="rerunsetup.cpp" line="57"/>
        <source>Cannot find setup script for </source>
        <translation type="unfinished"></translation>
    </message>
`

For German, the text after for must be a variable, German would be

> Setupskript für %1 kann nicht gefunden werden

Because you can't place %1 at the end in German Translation (grammatical issue)

**Comment by procount** [comment]
Ja ich weis.
@lurch is working on this to improve the parameterisation. Like I said before, it might be better to wait until this is finished because it will make the translation easier for you.

**Comment by ghost** [comment]
Okay, I'm going to wait until he's finished.

**Comment by lurch** [comment]
> @lurch is working on this to improve the parameterisation

...which in turn, is also waiting on me fixing #249 ... :sweat_smile: 

> QT linguist changed ts version. Should I change it back?

@procount what version of QT Linguist are you using? @RaspberryPiFan what version are you using?

**Comment by lurch** [comment]
And WRT the (translated) text not fitting in the GUI, can you check if you see the same behaviour in a (running) build of PINN with your updated translations included, or is this merely a display-bug in QT Linguist?

**Comment by ghost** [comment]
> And WRT the (translated) text not fitting in the GUI, can you check if you see the same behaviour in a (running) build of PINN with your updated translations included, or is this merely a display-bug in QT Linguist?

It's building, I'm going to ckeck after it's finished.


> > QT linguist changed ts version. Should I change it back?
> 
> @procount what version of QT Linguist are you using? @RaspberryPiFan what version are you using?

I'm using qt 5.11.3

**Comment by lurch** [comment]
> I'm using qt 5.11.3

Ahhh, I suspect that @procount is using QT Linguist 4, since QT4 is what PINN is built with. In which case you should probably revert the .ts version back to being qt4-compatible.

File: recovery/translation_de.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@
     <message>
         <location filename="backupthread.cpp" line="206"/>
         <source>%1: Checksumming (%2)</source>
-        <translation type="unfinished"></translation>
+        <translation></translation>
```
Comment (lurch) [inline recovery/translation_de.ts]: Empty translation??

**Comment by bovirus** [comment]
@lurch 

As I know the QT Linguist file edited with QT Linguist 5 are compatible and can be opened by QT Linguist 4.

The only difference is in the header 

QT Linguist 4
TS version="**_2.0_**" language="de" sourcelanguage="en"

QT Linguist 5
TS version="**_2.1_**" language="de" sourcelanguage="en"

But the TS version will be replaced at next PINN rebuild.

@RaspberryPiFan

You can use QT Linguist 5 or 4 as you like.

**Comment by procount** [comment]
Correct. But AFAIK it just changes the version of ts file from 2.0 to 2.1. Never seen any other differences.

**Comment by bovirus** [comment]
For all translator here you can find a list of QT-Linguist Editor version for Windows.

https://mega.nz/#F!KQhwUAKb!uaHNRZDdTK-c9HBdLN2XVQ

**_Qt-Linguist-x.xxx-platform-installer.exe_**
Installer (Setup version)

**_Qt-Linguist-x.xxx-platform-portable.zip_**
Portable (zip archive with several files) - Require archive decompression - Not required installation

**_Qt-Linguist-x.xxx-platform-standalone.exe_**
Portable (single exe file) - Not require decompression - Not required installation

Where **_platform_** could be **_x86_** (32bit) or **x64** (64bit).

32bit can run also on 64bit Windows OS.
64bit can run only on 64bit Windows OS.

@procount 

Please consider to add info translation in readme and also if you want this repository link.

**Comment by lurch** [comment]
> Correct. But AFAIK it just changes the version of ts file from 2.0 to 2.1. Never seen any other differences.

Ah, fair enough. But IMHO it's probably still worth deciding on which version you want to have stored in the repo, in order to avoid the TS version number ping-ponging back and forth depending on who last updated it? (or maybe I'm just being too pedantic? :man_shrugging:  )

**Comment by bovirus** [comment]
@lurch 

To have a match between TS version on TS repository you should use QT Linguist 4.8.x.

As I explained for me is not a problem because if I edit a file with TS vesrion 2.1 in the haeder PINN project can read it.

And when the developer run lupdate xxx.pro (project file name) it re-create TS 2.0 in all headers.
 
Added to my rewpository QT Linguist 4.8.7 standalone (single exe - no installation required - only one fiel).

https://mega.nz/#!WZQEVKbb!C2AMwC3_mslPaNcVDQkM-ylCL5J9ybhZ-v9TfYYzi5A

The best one to edit and save .ts PINN translation file.

**Comment by procount** [comment]
> (or maybe I'm just being too pedantic? 🤷‍♂ )

How could you say such a thing! :wink:

TBH, as I've never experienced a problem with differing .ts versions, I agree with @bovirus and I will accept either version. QT Linguist 4.8 (2.0) is preferred, but I will accept QT Linguist 5 (2.1) as well.
I'm just thankful that people are interested enough to do the translation!

File: recovery/translation_de.ts (hunk 1 above)
Comment (ghost) [reply recovery/translation_de.ts]: I'm going to fix it soon.

**Comment by ghost** [comment]
> And WRT the (translated) text not fitting in the GUI, can you check if you see the same behaviour in a (running) build of PINN with your updated translations included, or is this merely a display-bug in QT Linguist?

It's only a display bug in QT Linguist.



> For all translator here you can find a list of QT-Linguist Editor version for Windows.
> 
> https://mega.nz/#F!KQhwUAKb!uaHNRZDdTK-c9HBdLN2XVQ
> 
> **_Qt-Linguist-x.xxx-platform-installer.exe_**
> Installer (Setup version)
> 
> **_Qt-Linguist-x.xxx-platform-portable.zip_**
> Portable (zip archive with several files) - Require archive decompression - Not required installation
> 
> **_Qt-Linguist-x.xxx-platform-standalone.exe_**
> Portable (single exe file) - Not require decompression - Not required installation
> 
> Where **_platform_** could be **_x86_** (32bit) or **x64** (64bit).
> 
> 32bit can run also on 64bit Windows OS.
> 64bit can run only on 64bit Windows OS.
> 
> @procount
> 
> Please consider to add info translation in readme and also if you want this repository link.

I use Debian GNU/Linux.

**Comment by bovirus** [comment]
@RaspberryPiFan 
I believe that you installed **_qt4-dev-tools_**.

If you want when you will post the updated German language I will check it with qt/po check tools.

**Comment by ghost** [comment]
I included this in the buildroot fix.

**Pull request closed without merging**